src/box/user.cc

```diff
@@ -444,23 +463,38 @@ user_reload_privs(struct user *user)
 		if (it == NULL)
 			return -1;
 		IteratorGuard iter_guard(it);
-
+		bool in_rollback = stmt != NULL;
+		struct tuple *old_tuple = in_rollback ? stmt->old_tuple : NULL;
+		struct tuple *new_tuple = in_rollback ? stmt->new_tuple : NULL;
+		struct key_def *key_def = index->def->key_def;
+		int cmp_old_tuple = old_tuple != NULL ?
+			tuple_compare_with_key(old_tuple, HINT_NONE, key, 1,
+					       HINT_NONE, key_def) : 0;
+		int cmp_new_tuple = new_tuple != NULL ?
+			tuple_compare_with_key(new_tuple, HINT_NONE, key, 1,
+					       HINT_NONE, key_def) : 0;
```

Comment:
Nit: maybe we should do only one comparison in Release? I would put the second comparison to the scope of `#ifndef NDEBUG` or even get rid of it - it's quite reliable invariant that old and new tuple have the same primary key.

Reply:
Fixed: added an `ifdef` condition for the second comparison. I would leave the assertions for clarity of what's going on, since the code looks hacky.

Reply:
Well, now it looks weird - I think it's a bad idea to use `ifndef` right in the middle of an expression. Let's compare with key one of the tuples, which is not `NULL`, and  then compare two tuples with `tuple_compare` if they both are not NULL in the scope of `ifndef` with a comment like "just making sure that the tuples have the same primary key".

Reply:
Fixed as you suggested: removed the `ifndef` and moved the comparison under the assertion.